### PR TITLE
new: `Style Guide of Falco Rules` and `How to Share Your Falco Rules` pages

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind user-interface

/kind content

> /kind translation

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area blog

/area documentation

> /area videos

**What this PR does / why we need it**:

Add new
- Rules Style Guide
- How to Share Your Falco Rules

see https://github.com/falcosecurity/falco-website/issues/1009

@LucaGuerra @leogr @vjjmiras 
@loresuso and @jasondellaluce for additional feedback re correctness and completeness.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: content/en/docs/contribute/share-rules.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: How to Share Your Falco Rules
+description: Discover how to share and contribute your Falco rules
+weight: 20
+---
+
+
+To share your rule with the community, you can open a PR against the [rules repo](https://github.com/falcosecurity/rules). The PR review process ensures that the rules align with the project's best interests as per our [governance](https://github.com/falcosecurity/evolution/blob/main/GOVERNANCE.md), follow the [style guide](https://falco.org/docs/rules/style-guide/), and meet the additional requirements, including testing and maturity level outlined in the [rules repo](https://github.com/falcosecurity/rules) contributing guide. Reviewers will support you throughout the process.
+
```
Comment (incertum) [inline content/en/docs/contribute/share-rules.md:9]: Once we have the specific contributing guide in the rules repo, we can update the link.

File: content/en/docs/rules/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+In addition, the resources under [references/rules](https://falco.org/docs/reference/rules/) provide complementary information. We highly recommend regularly revisiting each guide to stay up-to-date with the latest advancements of Falco.
+
+
+### Naming
+
+Choose a concise title that summarizes the essence of the rule's purpose.
+
+### Description
+
+Aligning with Falco's rules maturity and adoption [framework](https://github.com/falcosecurity/rules/blob/main/proposals/20230605-rules-adoption-management-maturity-framework.md#rules-maturity-framework), it is encouraged to not just include a longer description of what the rule detects, but also to give advice on how to tune this rule and reduce possible noise. If applicable, elaborate on how to correlate the rule with other rules or data sources for incident response. However, keep them concise.
```
Comment (incertum) [inline content/en/docs/rules/style-guide.md:21]: Same once we have the specific contributing guide or rules maturity checklist doc in the rules repo, we can update the link. Meanwhile link to the proposal.

File: content/en/docs/contribute/share-rules.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+title: How to Share Your Falco Rules
+description: Discover how to share and contribute your Falco rules
+weight: 20
+---
+
+
+To share your rule with the community, you can open a PR against the [rules repo](https://github.com/falcosecurity/rules). The PR review process ensures that the rules align with the project's best interests as per our [governance](https://github.com/falcosecurity/evolution/blob/main/GOVERNANCE.md), follow the [style guide](/docs/rules/style-guide/), and meet the additional requirements, including testing and maturity level outlined in the [rules repo](https://github.com/falcosecurity/rules) contributing guide. Reviewers will support you throughout the process.
```
Comment (vjjmiras) [inline content/en/docs/contribute/share-rules.md:8]:
```suggestion
You can open a PR against the [rules repo](https://github.com/falcosecurity/rules) to share your rule with the community. The PR review process ensures that the rules align with the project's best interests as per our [governance](https://github.com/falcosecurity/evolution/blob/main/GOVERNANCE.md), follow the [style guide](/docs/rules/style-guide/), and meet the additional requirements, including testing and maturity level outlined in the [rules repo](https://github.com/falcosecurity/rules) contributing guide. Reviewers will support you throughout the process.
```

File: content/en/docs/rules/basic-elements.md [hunk 4]
```diff
@@ -250,3 +250,12 @@ In other words, visibility is defined in cascade and is quite important:
...
 - A rule can reference any macro.
+
+
+## Highly Relevant Blogposts and Resources
+
+- [Adaptive Syscalls Selection in Falco](https://falco.org/blog/adaptive-syscalls-selection/).
+- [Validating NIST Requirements with Falco](https://falco.org/blog/falco-nist-controls/).  
+- [PCI/DSS Controls with Falco](https://falco.org/blog/falco-pci-controls/).
+- [Style Guide of Falco Rules](/docs/contribute/style-guide/).
+- [Contributing](/docs/contribute/) page and [How to Share Your Falco Rules](/docs/contribute/share-rules/) guide.
```
Comment (vjjmiras) [inline content/en/docs/rules/basic-elements.md:261]: Are we sure this is the best place to introduce such advanced content? IMHO, most users would land there to learn the basics about rules, not to use them in such an advanced way.

File: content/en/docs/rules/style-guide.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+
+### Naming
+
+Choose a concise title that summarizes the essence of the rule's purpose.
+
+### Description
+
+Aligning with Falco's rules maturity and adoption [framework](https://github.com/falcosecurity/rules/blob/main/proposals/20230605-rules-adoption-management-maturity-framework.md#rules-maturity-framework), it is encouraged to not just include a longer description of what the rule detects, but also to give advice on how to tune this rule and reduce possible noise. If applicable, elaborate on how to correlate the rule with other rules or data sources for incident response. However, keep them concise.
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:35]:
```suggestion
Aligning with Falco's rules maturity and adoption [framework](https://github.com/falcosecurity/rules/blob/main/proposals/20230605-rules-adoption-management-maturity-framework.md#rules-maturity-framework), it is encouraged to not just include a longer description of what the rule detects but also to give advice on how to tune this rule and reduce possible noise. If applicable, elaborate on how to correlate the rule with other rules or data sources for incident response. However, keep them concise.
```

File: content/en/docs/rules/style-guide.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
+---
+title: Style Guide of Falco Rules
+description: Adopt best practices when writing and contributing Falco rules
+linktitle: Style Guide of Falco Rules
+weight: 90
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:5]:
```suggestion
weight: 95
```

File: content/en/docs/rules/style-guide.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+- macro: minerpool_https
+  condition: (fd.sport="443" and fd.sip.name in (https_miner_domains))
+...
+
+condition: ... and ((minerpool_http) or (minerpool_https) or (minerpool_other))
+```
+
+- Furthermore, it is preferred to use `and not` to consistently negate a positive sub-expression.
+- Avoid double-negation.
+
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:102]:
```suggestion
```

File: content/en/docs/rules/style-guide.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+When writing a rule for container workloads, you should include the fields we automatically fetch from the container runtime:
+
+```yaml
+output: container_id=%container.id image=%container.image.repository namespace=%k8s.ns.name pod_name=%k8s.pod.name
+```
+
+And as a general guidance, fields like `%proc.name`, `%proc.tty`, `%proc.cmdline` and fields related to process lineage such as `proc.aexepath[2]`, or fields around user information `%user.uid`, `%user.loginuid` are often of generic relevance to many rules. For specialized use cases, generic fields such as `%container.ip` or `%container.cni.json` can be further helpful for incident response, especially concerning non-network syscall related alerts in Kubernetes. These fields can be correlated, for example, with network proxy logs. 
+
+
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:129]:
```suggestion
```

File: content/en/docs/rules/style-guide.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+When considering upstreaming your rule, we expect the [Mitre Attack](https://attack.mitre.org/techniques/enterprise/) phase followed by the best Tactic or Technique, whichever is the best fit. This information is used to create an [overview document](https://github.com/falcosecurity/rules/blob/main/rules_inventory/rules_overview.md) of Falco's predefined rules and also help the Falco adoption process.
+
+Lastly, if the rule is relevant for a compliance use case, please add the corresponding `PCI_DSS_*` or `NIST_*` tag, referring to the [Validating NIST Requirements with Falco](https://falco.org/blog/falco-nist-controls/) and [PCI/DSS Controls with Falco](https://falco.org/blog/falco-pci-controls/) blog posts and rules contributing criteria outlined in the [rules repo](https://github.com/falcosecurity/rules).
+
+```yaml
+tags: [maturity_incubating, host, container, filesystem, mitre_defense_evasion, NIST_800-53_AU-10]
+```
+
+
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:157]:
```suggestion
```

File: content/en/docs/rules/style-guide.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+- macro: container
+  condition: (container.id != host)
+...
+
+- rule: Detect release_agent File Container Escapes
+...
+  condition: >
+    open_write and container and fd.name endswith release_agent and (user.uid=0 or thread.cap_effective contains CAP_DAC_OVERRIDE) and thread.cap_effective contains CAP_SYS_ADMIN
+...
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:63]:
```suggestion
- rule: Detect release_agent File Container Escapes
  ...
  condition: >
    open_write and container and fd.name endswith release_agent and (user.uid=0 or thread.cap_effective contains CAP_DAC_OVERRIDE) and thread.cap_effective contains CAP_SYS_ADMIN
...
```

File: content/en/docs/rules/style-guide.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+- macro: spawned_process
+  condition: (evt.type in (execve, execveat) and evt.dir=<)
+...
+
+- list: known_drop_and_execute_containers
+  items: []
+
+- rule: Drop and execute new binary in container
+...
+  condition: >
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:79]:
```suggestion
- rule: Drop and execute new binary in container
  ...
  condition: >
```

File: content/en/docs/contribute/share-rules.md [hunk 12]
```diff
@@ -0,0 +1,9 @@
+---
+title: How to Share Your Falco Rules
+description: Discover how to share and contribute your Falco rules
+weight: 20
+---
+
+
```
Comment (vjjmiras) [inline content/en/docs/contribute/share-rules.md:7]:
```suggestion
```

File: content/en/docs/rules/basic-elements.md [hunk 13]
```diff
@@ -250,3 +250,12 @@ In other words, visibility is defined in cascade and is quite important:
 - A macro can only reference macros defined before it.
 - A macro can reference any list.
 - A rule can reference any macro.
+
```
Comment (vjjmiras) [inline content/en/docs/rules/basic-elements.md:253]:
```suggestion
```

File: content/en/docs/rules/style-guide.md [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+  desc:
+  condition:
+  output:
+  priority:
+  tags:
+
+... other keys if applicable in no particular order
+
+```
+
```
Comment (vjjmiras) [inline content/en/docs/rules/style-guide.md:27]:
```suggestion
```

**Review by vjjmiras — CHANGES_REQUESTED** [review]

**Comment by vjjmiras** [comment]
It loos a great best practices guide. Thanks, @incertum!

**Comment by loresuso** [comment]
Melissa, thanks a lot for putting effort into this! This was something that we really needed!
I took a look at the content and for me it looks really good. Let's link this also in the README.md in the rules repository once it's merged!

**Comment by incertum** [comment]
@vjjmiras hope I got all change requests right. Thank you!

File: content/en/docs/rules/basic-elements.md (hunk 4 above)
Comment (incertum) [reply content/en/docs/rules/basic-elements.md]: Found a new place WDYT?

**Review by vjjmiras — DISMISSED** [review]
LGTM

File: content/en/docs/rules/style-guide.md [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+- To maintain performance, avoid mixing unrelated event types in one rule. Typically, only variants should be mixed together, for example: `evt.type in (open, openat, openat2)`.
+- The best practice and requirement for upstream rules are to only define positive `evt.type` expressions. Using `evt.type!=open`, for example, would imply each of the [supported syscalls](https://github.com/falcosecurity/libs/blob/master/driver/report.md), resulting in a significant performance penalty. For more information, read the [Adaptive Syscalls Selection in Falco](https://falco.org/blog/adaptive-syscalls-selection/) blog post.
+- After the `evt.type` filter, place your mainly positive filters to efficiently eliminate the most events step by step. An exception to this rule is the `container` macro, which can quickly eliminate many events. Therefore, the guiding principle of "divide and conquer" commonly used in database query recommendations, also applies to Falco's filter statements.
```
Comment (leogr) [inline content/en/docs/rules/style-guide.md]: Those are nice! However, they apply only to the `syscall` data source. Do we want to mention that?

File: content/en/docs/rules/style-guide.md [hunk 16]
```diff
@@ -0,0 +1,185 @@
...
+  priority:
+  tags:
+
+ ... other keys if applicable in no particular order
+
+```
+
+### Naming
+
+Choose a concise title that summarizes the essence of the rule's purpose.
```
Comment (leogr) [inline content/en/docs/rules/style-guide.md:30]:
```suggestion
Choose a concise title that summarizes the essence of the rule's purpose. Rule's name must start with an upper-case letter. For `macro` and `list`, use lowercase_separated_by_underscores, for example, `kernel_module_load`.
```

**Pull request merged**